org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/ast/RecordPattern.java

```diff
@@ -190,7 +190,7 @@ public TypeBinding resolveType(BlockScope scope, boolean isPatternVariable) {
 	}
 	private void setAccessorsPlusInfuseInferredType(BlockScope scope) {
 		this.isTotalTypeNode = super.coversType(this.resolvedType);
-		RecordComponentBinding[] components = this.resolvedType.components();
+		RecordComponentBinding[] components = this.resolvedType.capture(scope, this.sourceStart, this.sourceEnd).components();
```

Comment:
Rationale: accessing the components `r.ci()` would normally require to first evaluate `r` which includes capture conversion. This ensures that the components have real types, rather than wildcards (which are not types), given that the spec assumes that the components have types.

Reply:
Agree.. +1